src/processing/execution.rs

```diff
@@ -102,11 +114,13 @@ impl AsRenderOperations for RunSnippetOperation {
 
         let block_length = self.block_length.max(inner.max_line_length.saturating_add(1));
         for line in &inner.output_lines {
-            let chunks = line.chars().chunks(dimensions.columns as usize);
-            for chunk in &chunks {
-                operations.push(self.render_line(chunk.collect(), block_length));
-                operations.push(RenderOperation::RenderLineBreak);
-            }
+            // TODO: I am not sure how to approach this properly as I would be cutting through
```

Comment:
This is a good point I missed and will make this change a bit different I think. I'd suggest using this crate instead https://docs.rs/ansi-parser/latest/ansi_parser/ and parse each line with it and convert it into something that preserves the styles. For example, this could probably convert this into `TextBlock`s which you can then turn into render operations when you want to display them. Then, when you want turn that into `RenderOperation`s here, you would take the `dimensions.columns` as the max chunk size and if any line is bigger than you need to chunk it.

If this is more rust than you're comfortable (and dealing with the codebase, I'm actually not 100% sure this would work without changing anything about rendering strings) with, let me know and I get have a look. I'm starting to carve out a new release and I'd like this to be the last thing going in unless some issue comes up but I won't have time to work on this/the release until next week so no rush.

Reply:
Yeah, was a bummer! I didn't know it got chunked. I'll see what I can do, I have a few ideas for something that can maybe work but no promises as I have quite a few other things to do as well.
But seems like a fun challenge if it doesn't get too difficult.